src/halmos/__main__.py

```diff
@@ -856,13 +866,35 @@ def handle_assertion_violation(
             partial(
                 self._solve_end_to_end_callback,
                 ex=ex,
+                path_ctx=path_ctx,
                 description=description,
             )
         )
         self.submitted_futures.append(solve_future)
 
+    def _get_solver_output(self, future: Future, path_ctx: PathContext) -> SolverOutput:
+        path_id, query_file = path_ctx.path_id, str(path_ctx.dump_file)
+
+        if self.ctx.solving_ctx.executor.is_shutdown():
+            # if the thread pool is in the process of shutting down,
+            # we want to stop processing remaining models/timeouts/errors, etc.
+            e = "executor has been shutdown"
+            return SolverOutput.from_error(e, path_id=path_id, query_file=query_file)
+
+        if e := future.exception():
+            if not is_benign_solving_error(e):
+                error(f"encountered exception during assertion solving: {e!r}")
+            return SolverOutput.from_error(e, path_id=path_id, query_file=query_file)
+
+        try:
+            return future.result()
+        except Exception as e:
+            if not is_benign_solving_error(e):
+                error(f"encountered exception during assertion solving: {e!r}")
+            return SolverOutput.from_error(e, path_id=path_id, query_file=query_file)
+
```

Comment:
originally, other OSErrors were re-raised from the catch block, but now they aren't.  does this not alter behavior because exceptions raised from callbacks are silently ignored anyway?

Reply:
well I'm not sure why we re-raised, but it seems to me this is a good place to capture errors and keep track of errors